src/borg/compress.pyx

```diff
@@ -112,7 +112,7 @@ class LZ4(CompressorBase):
                 break
             if osize > 2 ** 30:
                 # this is insane, get out of here
-                raise Exception('lz4 decompress failed')
+                raise DecompressionError('lz4 decompress failed')
```

Comment:
Hmm so in case of corruption this would grow the buffer to >huge before aborting. Related https://github.com/lz4/lz4/issues/156